lib/src/store.dart

```diff
@@ -85,8 +85,10 @@ class Store extends Disposable {
       throw new StateError('Store has been disposed');
     }
 
-    return _stream.listen(onData,
```

Comment:
I don't like this pattern because:

1. It further complicates the question of object ownership
2. It further complicates the requirements for this store to be garbage collected.

That being said, I did find one instance of this in the DartSDK : https://github.com/dart-lang/sdk/blob/66b5248bf937ef1202f04b106c63730e57a008b8/sdk/lib/io/secure_server_socket.dart#L87